internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -279,62 +278,68 @@ func (a *Adapter) EnsureIntegrationPipelineRunsExist() (controller.OperationResu
 // EnsureGlobalCandidateImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
 // being updated when the Snapshot is created
 func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResult, error) {
-	if a.snapshot.GetLabels()[gitops.SnapshotTypeLabel] != gitops.SnapshotComponentType || !gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) {
-		a.logger.Info("The Snapshot wasn't created for a single component push event, not updating the global candidate list.")
+	if (a.snapshot.GetLabels()[gitops.SnapshotTypeLabel] != gitops.SnapshotComponentType ||
+		!gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot)) && !gitops.IsOverrideSnapshot(a.snapshot) {
+		a.logger.Info("The Snapshot was neither created for a single component push event nor override type, not updating the global candidate list.")
 		return controller.ContinueProcessing()
 	}
-	if gitops.IsSnapshotMarkedAsAddedToGlobalCandidateList(a.snapshot) {
-		a.logger.Info("The Snapshot's component was previously added to the global candidate list, skipping adding it.")
+
+	if gitops.IsOverrideSnapshot(a.snapshot) && !gitops.HaveAppStudioTestsSucceeded(a.snapshot) {
+		a.logger.Info("This is a override snapshot but hasn't yet passed all required integration tests, not updating the global candidate list.")
 		return controller.ContinueProcessing()
 	}
 
-	var componentToUpdate *applicationapiv1alpha1.Component
-	var err error
-	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
-		componentToUpdate, err = a.loader.GetComponentFromSnapshot(a.context, a.client, a.snapshot)
```

Comment:
I think we'll still need to get the component from Snapshot in case of `component` type Snapshots because for those kinds of Snapshots we only want to update that single component and leave the others as-is.

Reply:
Updated. Thanks.